### ISSUE TITLE
Is there a way to not insert a line feed like ejs?

### ISSUE THREAD

**Issue by muukii** [stencilproject/Stencil#89]
Is there a way to not insert a line feed like ejs? `-%>`

**Comment by kylef** [comment]
Not currently, but this is one of the things I'd like to support in Stencil 1.0. Haven't given though on the best way to solve this, potentially using `-` to indicate how whitespace and new lines should be treated just like Jinja2 or ejs.

The are many behaviours using both `+` and `-` to control whitespace in Jinja2 which you can find in their documentation http://jinja.pocoo.org/docs/dev/templates/#whitespace-control. I have been trying to keep the template language aligned with Jinja2 and Django as much as possible so the language is consistent with these similar existing language.

There has been previous discussion in #22, #32 and #85. I didn't like some of the other solutions because they change the behaviour for all tokens which may not be desired for some types of templates or tokens. For these reasons I was personally leaning towards supporting whitespace/newline control like Jinja2/ejs.

/cc @AliSoftware

**Comment by muukii** [comment]
@kylef 
Thank you for your reply.
I understood your idea.
Thank you for considering the correspondence.

**Comment by bejar37** [comment]
@kylef I started implementing Jinja2 behavior, and I think I'm pretty far along. What I've done so far is parse white-space controlling information ("{%+", "{%-", "-%}", and "+%}") for `Block` tokens, and used that information when constructing `TextNode`s during parse-time.

I have this working in "for", "if", and "ifnot" tags (I believe).
Before putting more time into this I want to make sure that you think I'm going down the right path. Some assumptions that I've made:
* the whitespace control characters "+" and "-" are only significant in `Block` tokens
* The only default tags where this makes sense are "if", "ifnot", and "for"
* Custom tags will have to change in order to opt-in to the whitespace-respecting behavior.

Let me know if you'd rather me just comment on this in a PR.

Thanks, and sorry if this is coming out of left field!

/cc @AliSoftware @muukii
